### ISSUE TITLE
[Feature request] Monitor the 'lag' on the nodes of a PXC a keep them in a new read group

### ISSUE THREAD

**Issue by davidducos** [Tusamarco/pxc_scheduler_handler#30]
I had been working with pxc_scheduler_handler, and I realized that when a node is in DESYNC mode, it goes to the bckHgR, which is ok, but if the node is up to date, it could serve reads, right?
That is why, what I'm thinking that could be useful is to monitor how behind a DESYNC node is, and if it is over a limit (similar to max_replication_lag), it will be moved to another group, like: 'useless node'.
You might be asking why I want this, well, if you set wsrep_sync_wait = 1 on the node, then you have consistent reads on that node, you could have higher latency per query, but it will be a consistent read without the need of add any new software. 
What do you think?

**Comment by Tusamarco** [comment]
@davidducos are u sure that with `wsrep_sync_wait = 1` and `wsrep_desync=1` you have still consistent reads? I think that if you decide to desync then the sync_wait do not apply. 
Said that I think that adding that functionality make sense anyhow. 
I am thinking to use the ` max_replication_lag: ` in `mysql_servers` table, plus a setting in the config specific for the cluster.
Let me think how to do it.

**Comment by davidducos** [comment]
@Tusamarco, I did some research and the reads are consistent, the query latency increases and there is no flow control (which is expected).

**Comment by Tusamarco** [comment]
For the record FR implemented and QA done successfully. I revisit my idea and will implement also UNIT test.
```
==============================================================================
                  TEST NAME                       RESULT  TIME (ms) COMMENT
------------------------------------------------------------------------------
[  1%] proxysql.reader_add1                      [ pass ]  61413
[  3%] proxysql.reader_add1_notdef_conf_hg       [ pass ]  60732
[  5%] proxysql.reader_add2                      [ pass ]  60821
[  7%] proxysql.reader_add3                      [ pass ]  60640
[  9%] proxysql.reader_add4                      [ pass ]  60808
[ 11%] proxysql.reader_desync1                   [ pass ]  61084
[ 13%] proxysql.reader_desync2                   [ pass ]  60834
[ 15%] proxysql.reader_desync3                   [ pass ]  60742
[ 17%] proxysql.reader_desync4                   [ pass ]  60960
[ 19%] proxysql.reader_kill1                     [ pass ]  244839
[ 21%] proxysql.reader_kill2                     [ pass ]  242455
[ 23%] proxysql.reader_kill3                     [ pass ]  242581
[ 25%] proxysql.reader_kill4                     [ pass ]  242024
[ 26%] proxysql.reader_reject_queries1           [ pass ]  61454
[ 28%] proxysql.reader_reject_queries2           [ pass ]  60738
[ 30%] proxysql.reader_reject_queries3           [ pass ]  61689
[ 32%] proxysql.reader_reject_queries4           [ pass ]  61037
[ 34%] proxysql.reader_to_maintenance1           [ pass ]  60748
[ 36%] proxysql.reader_to_maintenance2           [ pass ]  61509
[ 38%] proxysql.reader_to_maintenance3           [ pass ]  61007
[ 40%] proxysql.reader_to_maintenance4           [ pass ]  61299
[ 42%] proxysql.reader_to_readonly1              [ pass ]  61054
[ 44%] proxysql.reader_to_readonly2              [ pass ]  61087
[ 46%] proxysql.reader_to_readonly3              [ pass ]  61191
[ 48%] proxysql.reader_to_readonly4              [ pass ]  60983
[ 50%] proxysql.writer_add1                      [ pass ]  60614
[ 51%] proxysql.writer_add2                      [ pass ]  61248
[ 53%] proxysql.writer_add3                      [ pass ]  60936
[ 55%] proxysql.writer_add4                      [ pass ]  60757
[ 57%] proxysql.writer_desync1                   [ pass ]  60832
[ 59%] proxysql.writer_desync2                   [ pass ]  61140
[ 61%] proxysql.writer_desync3                   [ pass ]  60958
[ 63%] proxysql.writer_desync4                   [ pass ]  60825
[ 65%] proxysql.writer_kill1                     [ pass ]  241641
[ 67%] proxysql.writer_kill1_persist             [ pass ]  241606
[ 69%] proxysql.writer_kill2                     [ pass ]  244283
[ 71%] proxysql.writer_kill2b                    [ pass ]  242973
[ 73%] proxysql.writer_kill3                     [ pass ]  242007
[ 75%] proxysql.writer_kill4                     [ pass ]  243181
[ 76%] proxysql.writer_reject_queries1           [ pass ]  60791
[ 78%] proxysql.writer_reject_queries2           [ pass ]  61015
[ 80%] proxysql.writer_reject_queries3           [ pass ]  60674
[ 82%] proxysql.writer_reject_queries4           [ pass ]  60764
[ 84%] proxysql.writer_to_maintenance1           [ pass ]  60752
[ 86%] proxysql.writer_to_maintenance2           [ pass ]  60936
[ 88%] proxysql.writer_to_maintenance3           [ pass ]  60873
[ 90%] proxysql.writer_to_maintenance4           [ pass ]  61016
[ 92%] proxysql.writer_to_readonly1              [ pass ]  60996
[ 94%] proxysql.writer_to_readonly2              [ pass ]  61262
[ 96%] proxysql.writer_to_readonly3              [ pass ]  60832
[ 98%] proxysql.writer_to_readonly4              [ pass ]  61137
[100%] shutdown_report                           [ pass ]       
------------------------------------------------------------------------------
The servers were restarted 0 times
The servers were reinitialized 0 times
Spent 4927.778 of 5116 seconds executing testcases

Completed: All 52 tests were successful.
```

**Comment by davidducos** [comment]
@Tusamarco Thanks!! I'm going to be testing this next week in my local env and let you know if I found any issue. 
Best

**Comment by Tusamarco** [comment]
included in release 1.2.0

**Comment by davidducos** [comment]
@Tusamarco 
I was able to test it and it does what we expected. 
It was funny to see how the drop in the QPS when a node was removed from the read hostgroup when the load increased, and when it come back, the amount of QPS were higher than before.
I will start to blog about it tomorrow... hope to have it soon. 
Thanks again for all the work that you have done!
